packages/core/lib/commands/create/helpers.js

```diff
@@ -69,14 +70,28 @@ var Create = {
       );
     }
 
-    copy.file(from, to, function(err) {
+    copy.file(from, to, function (err) {
       if (err) return callback(err);
 
       replaceContents(to, templates.contract.name, name, callback);
+
+      let license;
+      const Config = require("@truffle/config");
+      try {
+        license = Config.detect(options).license;
+      } catch {}
+
+      if (!license) {
+        const packageJSONPath = path.join(process.cwd(), "package.json");
+        license = require(packageJSONPath).license;
```

Comment:
Don't we also need this in a `try/catch` in case the `package.json` doesn't exist? Then it will throw an error right?

Reply:
Yes, you are completely right. I've changed it. Thanks!